src/wallet/wallet.cpp

```diff
@@ -4050,3 +4052,114 @@ bool CMerkleTx::AcceptToMemoryPool(const CAmount& nAbsurdFee, CValidationState&
     return ::AcceptToMemoryPool(mempool, state, tx, nullptr /* pfMissingInputs */,
                                 nullptr /* plTxnReplaced */, false /* bypass_limits */, nAbsurdFee);
 }
+
+bool CWallet::IsSolvable(const CScript& script) const
+{
+    // This check is to make sure that the script we created can actually be solved for and signed by us
+    // if we were to have the private keys. This is just to make sure that the script is valid and that,
+    // if found in a transaction, we would still accept and relay that transaction. In particular,
+    // it will reject witness outputs that require signing with an uncompressed public key.
+    DummySignatureCreator creator(this);
+    SignatureData sigs;
+    return (ProduceSignature(creator, script, sigs) && VerifyScript(sigs.scriptSig, script, &sigs.scriptWitness, MANDATORY_SCRIPT_VERIFY_FLAGS | SCRIPT_VERIFY_WITNESS_PUBKEYTYPE, creator.Checker()));
+}
+
+CKeyID CWallet::GetKeyForDestination(const CTxDestination& dest) const
+{
+    // Only supports P2PKH, P2WPKH, P2SH-P2WPKH.
+    if (boost::get<CKeyID>(&dest)) {
+        return boost::get<CKeyID>(dest);
+    }
+    if (boost::get<WitnessV0KeyHash>(&dest)) {
+        return CKeyID(boost::get<WitnessV0KeyHash>(dest));
+    }
+    if (boost::get<CScriptID>(&dest)) {
+        CScript script;
+        CTxDestination inner_dest;
+        if (GetCScript(boost::get<CScriptID>(dest), script) && ExtractDestination(script, inner_dest)) {
+            if (boost::get<WitnessV0KeyHash>(&inner_dest)) {
+                return CKeyID(boost::get<WitnessV0KeyHash>(inner_dest));
+            }
+        }
+    }
+    return CKeyID();
+}
+
+static const std::string STYLE_STRING_NONE = "";
+static const std::string STYLE_STRING_LEGACY = "legacy";
+static const std::string STYLE_STRING_P2SH = "p2sh";
+static const std::string STYLE_STRING_SEGWIT = "segwit";
```

Comment:
The help for e.g. `getnewaddress` says `"p2wpkh"` etc. One or the other needs to be changed.

Reply:
Fixed, I think it's consistent now.